lighthouse-core/gather/gather-runner.js

```diff
@@ -458,6 +469,16 @@ class GatherRunner {
           passConfig,
           // *pass() functions and gatherers can push to this warnings array.
           LighthouseRunWarnings: baseArtifacts.LighthouseRunWarnings,
+          /** @param {keyof LH.GathererArtifacts} name */
+          async dangerouslyUsePreviousArtifact(name) {
```

Comment:
feels like this fn should be defined outside of this for loop.

Reply:
what would be your preferred flavor?

`dangerouslyUsePreviousArtifact: GatherRunner.dangerouslyUsePreviousArtifact.bind(gathererResults)`
`dangerouslyUsePreviousArtifact: GatherRunner.createPreviousArtifactFn(gathererResults)`
`dangerouslyUsePreviousArtifact(name) { return GatherRunner.dangerouslyUsePreviousArtifact(gathererResults, name); }`

Reply:
`dangerouslyUsePreviousArtifact: GatherRunner.dangerouslyUsePreviousArtifact.bind(gathererResults)`